### ISSUE TITLE
Negative evaluation metric (classification_error)

### ISSUE THREAD

**Issue by frankibem** [Microsoft/CNTK#2505]
I'm training a model using cross_entropy_with_softmax as the loss and classification_error for the evaluation metric. After a few minibatches, the metric was calculated as -219100133361550.00 % x 102 after which the metric is fixed at 0.00% x <no_of_samples>. However, the loss is still slowly but consistently decreasing.

Considering how classification_error is calculated, should this be possible? Do I abort the training at this point?

**Comment by cha-zhang** [comment]
This indeed looks strange. Are you doing distributed training?

**Comment by frankibem** [comment]
No. I'm running on a single machine with 1 GPU.

**Comment by cha-zhang** [comment]
We need more details to track down the issue. Could you share a small repro?

**Comment by frankibem** [comment]
[Here's](https://github.com/frankibem/asllvd_issue/tree/master) a repo with the training script and links to download the dataset. I haven't been able to reproduce the exact scenario described above, but the summary stats. for the loss and metric are occasionally computed to be very large.

**Comment by cha-zhang** [comment]
@frankibem has the issue been resolved?

**Comment by frankibem** [comment]
I've tried to replicate it several times but no luck, so closing for now.

**Comment by cha-zhang** [comment]
By "replicate it several times but no luck", you mean the metrics return to normal? Or it's still wrong and you can't fix it? If the latter, let's keep this issue open.

**Comment by frankibem** [comment]
I've been unable to get the metrics to go largely negative like before. I'll update if it happens again.

**Comment by cha-zhang** [comment]
Great. Then we will close this. Thanks!

**Comment by frankibem** [comment]
The issue is popping up again so I'm reopening. Please ignore the previous url. You can access a repo with the necessary details at https://github.com/frankibem/cntk-issue.

**Comment by frankibem** [comment]
@cha-zhang I have updated my [repository](https://github.com/frankibem/cntk-issue) and added a log file that shows the issue. It is consistently reoccurring and may be related to #2691.

**Comment by bartdotnet** [comment]
Hi,
Unable to test my models on an Azure Standard NC6s_v2 (6 vcpus, 112 GB memory) with Data Science Virtual Machine (DSVM)' is a 'Windows Server 2016'

Need I to update the Nvidia Drivers due to Meltdown / Spectre?

Has this been reproduced by Microsoft?

Thanks in advance,

**Comment by ebarsoumMS** [comment]
@bartdotnet  can you open different issue, this is unrelated to this issue.

**Comment by KeDengMS** [comment]
Can you run uname -a to find out your VM kernel version? if it's 4.13.0-1005-azure, then you'll need NVidia driver 384.111

**Comment by bartdotnet** [comment]
Done: https://github.com/Microsoft/CNTK/issues/2845


Sent from Mail<https://go.microsoft.com/fwlink/?LinkId=550986> for Windows 10

________________________________

**Comment by frankibem** [comment]
Not reoccurring since release 2.5
